src/components/services-view/services-view.controller.ts

```diff
@@ -1,156 +1,173 @@
-        subCats = subCats.concat(categoryObj.subCategories);
-      }
-    });
-
-    subCats = this.makeRows(subCats);
-
-    return subCats;
-  };
-
-  public handleClick(item: any, e: any) {
-    item.ctrl.serviceToOrder = item;
-    item.ctrl.openOrderingPanel();
-  };
-
-  public openOrderingPanel() {
-    this.ctrl.orderingPanelvisible = true;
-  };
-
-  public closeOrderingPanel = () => {
-    this.ctrl.orderingPanelvisible = false;
-  };
-
-  private updateAll() {
-    this.updateCategories();
-    this.updateServices();
-  }
-
-  private updateServices() {
-    this.ctrl.origServices = this.normalizeServices();
-    this.ctrl.filteredServices = this.ctrl.origServices;
-  }
-
-  private updateCategories() {
-    this.ctrl.categories = this.ctrl.categories;
-    this.ctrl.subCategories = this.getSubCategories('all');
-  }
-
-  private makeRows(subCats: any) {
-    let subCatsRows: any = [];
-    for (let i = 0, len = subCats.length; i < len; i += this.subCatMaxRowLength) {
-      subCatsRows.push(subCats.slice(i, i + this.subCatMaxRowLength));
+
+    private updateCatalogServices() {
+        this.ctrl.allItems = this.ctrl.allItems.concat(this.normalizeData('service', this.ctrl.catalogServices));
+        this.updateState();
+    }
+
+    private updateOpenshiftImages() {
+        this.ctrl.allItems = this.ctrl.allItems.concat(this.normalizeData('image', this.ctrl.openshiftImageStreams));
+        this.updateState();
     }
-    return subCatsRows;
-  }
-
-  private getRowOfSubCategory(subCategory: string) {
-    for (let row = 0; row < this.ctrl.subCategories.length; row += 1) {
-      for (let card = 0; card < this.ctrl.subCategories[row].length; card += 1) {
-        if (this.ctrl.subCategories[row][card].id === subCategory) {
-          return row;
+
+    private makeRows(subCats: any) {
+        let subCatsRows: any = [];
+        for (let i = 0, len = subCats.length; i < len; i += this.subCatMaxRowLength) {
+            subCatsRows.push(subCats.slice(i, i + this.subCatMaxRowLength));
         }
-      }
+        return subCatsRows;
+    }
+
+    private getRowOfSubCategory(subCategory: string) {
+        for (let row = 0; row < this.ctrl.subCategories.length; row += 1) {
+            for (let card = 0; card < this.ctrl.subCategories[row].length; card += 1) {
+                if (this.ctrl.subCategories[row][card].id === subCategory) {
+                    return row;
+                }
+            }
+        }
+    }
+
+    private normalizeData(type: string, items: any) {
+        let retSvcs = [];
+        let objClass: any;
+        _.each(items, (item: any, key: string) => {
+            if (type === 'service') {
+                objClass = new CatalogServiceItem(item);
+            } else if (type === 'image') {
+                objClass = new OpenshiftImageItem(item);
+            }
+            // each item/card needs a call back to 'this' class
+            _.assign(objClass, {ctrl: this});
```

Comment:
I see, I'd rather pass references to functions than the entire controller.  How many functions is it needing access to?